components/script/parse/xml.rs

```diff
@@ -33,6 +34,18 @@ impl<'a> TreeSink for servoxmlparser::Sink {
 
     fn parse_error(&mut self, msg: Cow<'static, str>) {
         debug!("Parse error: {}", msg);
+
+        // Remove all elements from document
+        while let Some(node) = self.document.GetDocumentElement() {
```

Comment:
I would rather create a new document than clear the current one, but there is no way to get the newly created document back to `ParseFromString`. My other concern is that parsing may continue after a `parser_error` is triggered.